.github/workflows/upload-release-builds.yml

```diff
@@ -0,0 +1,48 @@
+name: Upload Release Builds
+
+
+# When a release occurs, download artifacts published from build.yml,
+# append the release tag to their names, and upload them to the release.
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+
+jobs:
+  upload-release-builds:
+    name: Upload release builds
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # release changes require contents write
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Download artifacts from commit ${{ github.sha }}
+        uses: dawidd6/action-download-artifact@v3
+        with:
+          workflow: build.yml
+          workflow_conclusion: success
+          commit: ${{ github.sha }}
+          skip_unpack: true
+          path: dist
+
+      - name: Print dist directory
+        run: ls -R dist
+
+      - name: Upload release builds
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        # We want to append the release tag to the name before upload
+        run: |
+          for file in ${{ github.workspace }}/dist/*; do
+            if [ -f "$file" ]; then
+              FILE_NAME=$(basename "$file")
```

Comment:
[consider] Odd mixture of uppercase/lowercase (e.g. `file` vs `FILE_NAME`). Recommend lowercase - there's no particular significance of these uppercase variables.

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/pull/1241/commits/00d236b4775cd8fbf16c76876090888a5132dcde

Always just assumed this was a bash style thing.

Reply:
I think I mostly see uppercase for environment variables, rather than variables local to a script.